clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -417,10 +416,96 @@ SYCLToolChain::SYCLToolChain(const Driver &D, const llvm::Triple &Triple,
   getProgramPaths().push_back(getDriver().Dir);
 }
 
+static bool doLLVMBCEmit(llvm::opt::ArgStringList &CC1Args) {
+  for (auto StrArg : CC1Args)
+    if (std::string(StrArg) == "-emit-llvm-bc")
+      return true;
+  return false;
+}
+
+void SYCLToolChain::AddSYCLDeviceLibs(const llvm::opt::ArgList &Args,
+                                      llvm::opt::ArgStringList &CC1Args) const {
+  enum SYCLDeviceLibType { sycl_devicelib_wrapper, sycl_devicelib_fallback };
+  struct DeviceLibOptInfo {
+    StringRef devicelib_name;
+    StringRef devicelib_option;
+  };
+
+  bool NoDeviceLibs = false;
+  // Currently, libc, libm-fp32 will be linked in by default. In order
+  // to use libm-fp64, -fsycl-device-lib=libm-fp64/all should be used.
```

Comment:
Do you see any problems with linking `libm-fp64` by default?

Reply:
Hi, @bader 
I think it should be OK to link libm-fp64 by default too. I will find a some machine without fp64 extension to see if any problem exposed.
Thanks very much.

Reply:
There should be no problems. If math functions with doubles are not used there will be no unsupported functions in the code.

Reply:
@bader @mdtoguchi @vzakhari 
Have updated the patch to link libm-fp64 by default. But I suggest to re-discuss the behavior of "-fsycl-device-lib=..."
Currently, "-fsycl-device-lib=.." won't impact our "default" link behavior, it seems that current behavior of "-fsycl-device-lib=" is useless after we link all device libraries by default.
Maybe aligning behaviors of "-fsycl-device-lib" and "-fno-sycl-device-lib" is more reasonable, if user addes "-fsycl-device-lib=libm-fp32", we won't do "default" link, instead, we only link libm-fp32 device libraries, the same to libc, libm-fp64...
And I think we should re-consider the "-fsycl-device-lib=..." no matter we choose this PR or https://github.com/intel/llvm/pull/2783 in the end.
Thanks very much.

Reply:
I would say we have just one option - `-fsycl` analog for `-nostdlib` i.e. `-fno-sycl-device-lib`.
I don't think we really need a fine grained control over which part of standard library is linked.

Reply:
Agree with @bader. We shouldn't modify what we link by default with an option that adjusts the adding of libs.  Similar to adding `-lgcc` on the command line, that doesn't override the default libs already linked in.  A single option that disables all is reasonable.